qcodes/sphinx_extensions/parse_parameter_attr.py

```diff
@@ -1,215 +1,18 @@
-    """
-    Try to extract an attribute as a proxy object with a repr containing the code
-    if the class the attribute is bound to is a subclass of ``InstrumentBase``
-    and the attribute is not private.
-
-    Args:
-        object_to_document_attr_on: The type (not instance of the object to detect the
-            attribute on.
-        name: Name of the attribute to look for.
-        *default: Default obejct to use as a replacement if the attribute could not be
-            found.
-
-    Returns:
-        Attribute looked up, proxy object containing the code of the attribute as a
-        repr or a default object.
-    """
-    if (
-        inspect.isclass(object_to_document_attr_on)
-        and issubclass(object_to_document_attr_on, InstrumentBase)
-        and not name.startswith("_")
-    ):
-        try:
-            attr = safe_getattr(object_to_document_attr_on, name)
-        except AttributeError:
-            LOGGER.debug(
-                f"Attempting to load attribute {name} on "
-                f"{object_to_document_attr_on} via parsing"
-            )
-            mro = inspect.getmro(object_to_document_attr_on)
-            attr = None
-            for classobj in mro:
-                try:
-                    param_dict = eval_params_from_code(
-                        inspect.getsource(classobj), classobj.__name__
-                    )
-                    if param_dict.get(name) is not None:
-                        attr = param_dict[name]
-                        break
-                except TypeError:
-                    continue
-            if attr is None:
-                LOGGER.debug(
-                    f"Falling back to default Sphinx attribute loader for {name}"
-                    f" on {object_to_document_attr_on}"
-                )
-                attr = safe_getattr(object_to_document_attr_on, name, *default)
-    else:
-        attr = safe_getattr(object_to_document_attr_on, name, *default)
-    return attr
-
-
-def setup(app: Any) -> Dict[str, Union[str, bool]]:
-    """Called by sphinx to setup the extension."""
-    app.setup_extension("sphinx.ext.autodoc")  # Require autodoc extension
-
-    app.add_autodoc_attrgetter(object, qcodes_parameter_attr_getter)
-
-    return {
-        "version": "0.1",
-        "parallel_read_safe": True,  # Not tested, should not be an issue
-        "parallel_write_safe": True,  # Not tested, should not be an issue
-    }
+from qcodes.extensions.sphinx_extension.parse_parameter_attr import (
+    ParameterProxy,
+    eval_params_from_code,
+    extract_code_as_repr,
+    extract_statements_from_node,
+    find_class,
+    find_init_func,
+    parse_init_function_from_str,
+    qcodes_parameter_attr_getter,
+    setup,
+)
+from qcodes.utils import issue_deprecation_warning
+
+issue_deprecation_warning(
+    "QCoDeS Sphinx extension: qcodes_parameter_attr_getter has been removed "
+    "from the public API. Please raise an issue at github.com/qcodes/qcodes "
```

Comment:
@caenrigen do you use the sphinx extension in your projects that depends on qcodes for better docs generation for parameters?